### ISSUE TITLE
Playlist Search Feature

### ISSUE THREAD

**Issue by JPili** [ytmdesktop/ytmdesktop#676]
- [X] I understand that **YTMDesktop have NO affiliation with Google or YouTube**.
- [X] I verified that there is no open issue for the same subject.

**Is your feature request related to a problem? Please describe.**
Considering that people have been asking for this feature from YouTube for years now, I think it's safe to say they aren't going to add it anytime soon. Unless the song is near the top of the playlist or the playlist is small, not having this feature prevents us from starting a playlist from a certain song, removing a song we're not interested in, etc. I don't even know if it's possible to implement this feature since the app essentially just fetches the YouTube Music site...

**Describe the solution you'd like**
It would be preferable if the feature worked exactly like or better than Spotify's playlist search feature. So it'll allow us to search for a song to:
1) Create a makeshift playlist within the playlist using the search term as a filter (e.g., have the search term be "John Mayer" so the playlist will only play songs that have the term "John Mayer" in them.). Then, have the option to remove the search term so every song will show up in the queue again.
2) Delete the song from the playlist.

But honestly, I'd be happy with just being able to search for a song and deleting it from the playlist.

**Describe alternatives you've considered**
I can't think of any alternatives at the moment.

**Additional context**

**Comment by NovusTheory** [comment]
This might be possible... but it's more of an annoyance to do. Every playlist has to be downloaded to your system and somehow manage to keep updated as well (constant querying/other) which I'd imagine is not ideal especially because YouTube uses a cursor based query (which is why YTM also doesn't have proper scrolling like Spotify) making it an extraordinarily long process to index everything.

In short, it could be done, but long term is an annoyance to make and maintain.

**Comment by JPili** [comment]
Oh ok, I understand. I guess all I can do is hope YouTube somehow decides to focus more on quality of life in the future. Thanks for the reply and explanation :)
